### PR TITLE
fix(kubernetes/diff): check correctly & ignore apiVersion

### PR DESCRIPTION
**Pull request by sh0rez**
version check was broken, now auto choose should work correctly

ignores apiVersion, too messy for now

### CODE REVIEW

File: pkg/kubernetes/subsetdiff.go [hunk 1]
```diff
@@ -32,59 +37,85 @@ func (k Kubectl) SubsetDiff(y string) (string, error) {
...
+	}
 
-		is, err := yaml.Marshal(subset(m, rawIs))
-		if err != nil {
-			return "", err
+	for rs > 0 {
+		select {
+		case d := <-results:
+			docs = append(docs, d)
+			rs--
```
Comment (gouthamve) [inline pkg/kubernetes/subsetdiff.go]: We should decrement it in errors also? More like `for i := 0; i < rs; i++` would be better.

File: pkg/kubernetes/subsetdiff.go [hunk 2]
```diff
@@ -13,14 +13,19 @@ import (
...
 	live, merged string
 }
 
 func (k Kubectl) SubsetDiff(y string) (string, error) {
-	docs := map[string]difference{}
+	docs := []difference{}
 	d := yaml.NewDecoder(strings.NewReader(y))
-	for {
 
+	rs := 0
```
Comment (gouthamve) [inline pkg/kubernetes/subsetdiff.go]: `numResults`?

File: pkg/kubernetes/subsetdiff.go [hunk 3]
```diff
@@ -13,14 +13,19 @@ import (
...
+	errs := make(chan error)
```
Comment (gouthamve) [inline pkg/kubernetes/subsetdiff.go]: `errCh`?

File: pkg/kubernetes/subsetdiff.go [hunk 4]
```diff
@@ -13,14 +13,19 @@ import (
...
+	results := make(chan difference)
```
Comment (gouthamve) [inline pkg/kubernetes/subsetdiff.go]: `resultsCh`?

File: pkg/kubernetes/subsetdiff.go [hunk 5]
```diff
@@ -32,59 +37,85 @@ func (k Kubectl) SubsetDiff(y string) (string, error) {
...
 		}
-		if string(is) == "{}\n" {
-			is = []byte("")
-		}
-		docs[name] = difference{string(is), string(should)}
 	}
+	close(results)
+	close(errs)
 
 	s := ""
```
Comment (gouthamve) [inline pkg/kubernetes/subsetdiff.go]: What is `s`?

File: pkg/kubernetes/subsetdiff.go [hunk 6]
```diff
@@ -32,59 +37,85 @@ func (k Kubectl) SubsetDiff(y string) (string, error) {
...
-	for k, v := range docs {
-		d, err := diff(k, v.live, v.merged)
+	for _, d := range docs {
+		sd, err := diff(d.name, d.live, d.merged)
```
Comment (gouthamve) [inline pkg/kubernetes/subsetdiff.go]: `sd`?

File: pkg/kubernetes/subsetdiff.go (hunk 5 above)
Comment (sh0rez) [reply pkg/kubernetes/subsetdiff.go]: `s` is the final output string that every diff we have is appended to

**Pull request merged**